synapse/handlers/receipts.py

```diff
@@ -178,35 +183,23 @@ def filter_out_hidden(events: List[JsonDict], user_id: str) -> List[JsonDict]:
 
             for event_id in content.keys():
                 event_content = content.get(event_id, {})
-                m_read = event_content.get(ReceiptTypes.READ, {})
 
-                # If m_read is missing copy over the original event_content as there is nothing to process here
-                if not m_read:
-                    new_event["content"][event_id] = event_content.copy()
+                m_read = event_content.get(ReceiptTypes.READ, None)
+                if m_read:
+                    new_event["content"][event_id] = {ReceiptTypes.READ: m_read}
                     continue
 
-                new_users = {}
-                for rr_user_id, user_rr in m_read.items():
-                    try:
-                        hidden = user_rr.get("hidden")
-                    except AttributeError:
-                        # Due to https://github.com/matrix-org/synapse/issues/10376
-                        # there are cases where user_rr is a string, in those cases
-                        # we just ignore the read receipt
-                        continue
+                # Filter the private read receipts to only the requesting user
+                m_read_private = event_content.get(ReceiptTypes.READ_PRIVATE, None)
+                if m_read_private:
+                    user_rr = m_read_private.get(user_id, None)
+                    if user_rr:
+                        new_event["content"][event_id] = {
+                            ReceiptTypes.READ_PRIVATE: {user_id: user_rr.copy()}
+                        }
```

Comment:
I think this will mean that if a user has a private read receipt at event `A`, then we'll not return any other user's at that event.

We likely want:

```suggestion
                        new_event["content"].setdefault(event_id, {})[ReceiptTypes.READ_PRIVATE] = {user_id: user_rr.copy()}
```

This would be a great thing to add a unit test for!

Reply:
You're right, this could cause an issue, I am trying to figure out the optimal solution

Reply:
Feels to me like we would definitely want a test with multiple receipt types on the same event too, since this is the first time we're doing it.